pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -121,6 +121,15 @@ func (i *objectCacheItem) startReflector() {
 	i.waitGroup.Wait()
 	i.waitGroup.Add(1)
 	defer i.waitGroup.Done()
+	resourceVersion := ""
+	// store has only a single item at most
+	for _, obj := range i.store.List() {
+		if item, ok := obj.(metav1.Object); ok {
+			resourceVersion = item.GetResourceVersion()
+		}
+	}
+	// Avoid "Too large resource version" error caused by bookmark
+	i.reflector.SetLastSyncResourceVersion(resourceVersion)
```

Comment:
resetting to the object resourceVersion risks triggering a "too old / gone" error.

what does the reflector do in the case of a "Too large resource version" error?

what does the reflector do in the case of a "Too old / gone" error?

Reply:
> resetting to the object resourceVersion risks triggering a "too old / gone" error.

It shouldn't - the semantic that we're using is "NotOlderThan", so passing very old version isn't necessary bad (and in many cases it will actually be really old RV).

No matter what, we need a comprehensive tests to actually avoid future regressions.

> what does the reflector do in the case of a "Too large resource version" error?

We're effectively relisting from resourceVersion=""
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L525

> what does the reflector do in the case of a "Too old / gone" error?

if it's coming from list, the handler is the same as for too large:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L517

if it's coming from watch - we go back to relist (first from existing RV):
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L444

Reply:
@wojtek-t  It will retry 10 times before relisting from resourceVersion="", each attempt will take 3 seconds. So pod starting delayed for 30 seconds.

Reply:
> the semantic that we're using is "NotOlderThan", so passing very old version isn't necessary bad

if that's true, then I don't understand what issue this PR is trying to solve... wouldn't passing the old secret or bookmark version we saw last work properly?